clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -8353,8 +8353,6 @@ void FileTableTform::ConstructJob(Compilation &C, const JobAction &JA,
       addArgs(CmdArgs, TCArgs, {Arg});
       break;
     }
-    default:
-      llvm_unreachable("unknown file table transformation kind");
```

Comment:
Could you please explain this one?   Are we losing the "unreachable" message for the cases not handled in the switch?

Reply:
Clang knows that the values of the switch can only be the ones listed in the cases before the default, and it gives the error that I am mentioning above.

Reply:
Ah, got it!   Sorry, I didn't realize @elizabethandrews was asking about the same label.

Reply:
If we want to add, the unreachable to make sure we catch a potential wrong value of the switch then, we need to rewrite the switch statement all together. May be asserting that Tf.TheKind can only be the values listed and put that assertion before the switch statement. Let me know if you think that this is better approach.

Reply:
exactly - I was also curious and verified that